flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/schema/FlinkRelOptTable.scala

```diff
@@ -221,12 +230,40 @@ class FlinkRelOptTable protected(
     val cluster: RelOptCluster = context.getCluster
     if (table.isInstanceOf[TranslatableTable]) {
       table.asInstanceOf[TranslatableTable].toRel(context, this)
-    } else if (Hook.ENABLE_BINDABLE.get(false)) {
-      LogicalTableScan.create(cluster, this)
-    } else if (CalcitePrepareImpl.ENABLE_ENUMERABLE) {
-      EnumerableTableScan.create(cluster, this)
     } else {
-      throw new AssertionError
+      if (!context.isInstanceOf[FlinkToRelContext]) {
+        // If the transform comes from a RelOptRule,
+        // returns the scan directly.
+        LogicalTableScan.create(cluster, this)
+      } else {
+        // Get row type of physical fields.
+        val physicalFields = getRowType.getFieldList
```

Comment:
If you are warping lines, make each call a separate line, like:
>val physicalFields = getRowType
            .getFieldList
            .filter(f => !columnExprs.contains(f.getName))
            .toList`